src/components/TaskSupplier.js

```diff
@@ -0,0 +1,123 @@
+import {useState} from 'react';
+import AddTask from './AddTask';
+import Tasks from './Tasks';
+import Footer from './Footer';
+import './TaskSupplier.css';
+
+import {generateUniqueID} from "web-vitals/dist/modules/lib/generateUniqueID";
+import {useCollectionData} from "react-firebase-hooks/firestore";
+import { collection, deleteDoc, doc, query, serverTimestamp, setDoc, orderBy, updateDoc } from "firebase/firestore";
+
+const collectionName = "List-Collection";
+const subCollectionName = "Tasks-Collection";
+
+export default function TaskSupplier(props) {
+
+    const [showComplete, setShowComplete] = useState(false);
+    const [sortBy, setSortBy] = useState("created");
+
+    const qTask = query(collection(props.db, collectionName, props.currentListId, subCollectionName), orderBy(sortBy))
```

Comment:
Lots of duplication of:
props.db, collectionName, props.currentListId, subCollectionName
How about:
const listCollection = collection(props.db, collectionName, props.currentListId, subCollectionName)
and then you can replace all instances with listCollection